caddyhttp/httpserver/server.go

```diff
@@ -92,6 +92,10 @@ func NewServer(addr string, group []*SiteConfig) (*Server, error) {
 	if err != nil {
 		return nil, err
 	}
+	// Since Go 1.7 HTTP/2 is enabled only if TLSConfig.NextProtos includes the string "h2".
+	if HTTP2 && s.Server.TLSConfig != nil && len(s.Server.TLSConfig.NextProtos) == 0 {
+		s.Server.TLSConfig.NextProtos = []string{"h2"}
+	}
```

Comment:
At first I was thinking this logic should go into the `caddytls.MakeTLSConfig()` function, but this pertains specifically to HTTP, not all TLS. So I think this is actually in the right place. :+1: Setting "h2" for another type of server might mess things up.